nomad/structs/structs.go

```diff
@@ -4554,6 +4594,7 @@ func (j *Job) Copy() *Job {
 	nj.Constraints = CopySliceConstraints(nj.Constraints)
 	nj.Affinities = CopySliceAffinities(nj.Affinities)
 	nj.Multiregion = nj.Multiregion.Copy()
```

Comment:
I don't understand the wider go context here, but I was just trying to follow the Multiregion/Affinities/Constraints/Datacenters convention. I think this may be deep-copying the UI object after the `*nj = *j` shallow copied it? At least that's how I read it.

Reply:
`nj.Multiregion = nj.Multiregion.Copy()` this line is not copying anything, you are assigning the value of nj.Multiregion to nj.Multiregion, it self. You need to return a copy of j, so it should be 

```suggestion
	nj.Multiregion = j.Multiregion.Copy()
```

Reply:
Hope I've addressed this [here](https://github.com/hashicorp/nomad/pull/18292/commits/1e7dd3949e989f826e39f58af4d25ab87a0bc93b)